mlfinlab/labeling/matrix_flags.py

```diff
@@ -0,0 +1,110 @@
+                        lowest row etc. THe proportion of points in each decile is reported to finalize the column.
+                        The first tenth of the data is transformed to the leftmost column, the next
+                        tenth to the next column on the right and so on until finally a 10 by 10 matrix is achieved.
+        """
+        # The relevant data to create matrix for current day consists of the day's return and the window number of days
+        # preceding.
+        data_window = self.data[row_num - window: row_num]
+
+        # Find values for cutoff percentiles (deciles) over the entire data window. percentile_cutoffs is a list of
+        # cutoffs for the 10th, 20th, 30th etc. percentiles in the data window.
+        percentile_cutoffs = []
+        for i in [10, 20, 30, 40, 50, 60, 70, 80, 90, 100]:
+            percentile_cutoffs.append(np.percentile(data_window, i))
+
+        # Each value in bins corresponds to the value of the same index in data_window. If bin is 0, the corresponding
+        # point in data_window is in the 0-10th percentile, if 1 then 10-20th percentile, ..., if 9 then 90-100th
+        # percentile.
+        bins = np.digitize(data_window, bins=percentile_cutoffs, right=True)
+
+        # Dictionary to map values in data window to their percentile bin.
+        value_to_bin = dict(zip(data_window, bins))
+
+        # Split data window into 10 chronological sub-windows.
+        data_split = np.array_split(data_window, 10)
+
+        # Create the matrix, one column at a time from left to right. The leftmost column corresponds to the first
+        # tenth of data, and the rightmost corresponds to the final tenths. The top element of each column corresponds
+        # to the count in the tenth in the 90-100th percentile of data window, second from top is count in the tenth
+        # in the 80-90th percentile and so on until the last element is the count in the 0-10th percentile.
+        matrix = pd.DataFrame()
+        col_num = 0  # Left most column (out of 10) to populate using the first tenth of data.
+        for tenth in data_split:
+            # Apply the dictionary to data to get the bins each data point belongs to.
+            tenth_bins = np.vectorize(value_to_bin.get)(tenth)
+            # We count the number of times each bin appears in each tenth of data, starting from the highest decile.
+            column = [np.count_nonzero(tenth_bins == 9), np.count_nonzero(tenth_bins == 8),
+                      np.count_nonzero(tenth_bins == 7), np.count_nonzero(tenth_bins == 6),
+                      np.count_nonzero(tenth_bins == 5), np.count_nonzero(tenth_bins == 4),
+                      np.count_nonzero(tenth_bins == 3), np.count_nonzero(tenth_bins == 2),
+                      np.count_nonzero(tenth_bins == 1), np.count_nonzero(tenth_bins == 0)]
+            matrix[col_num] = column
```

Comment:
This part can be shorter:
<pre>
        column, _ = np.histogram(tenth_bins, bins=[i for i in range(11)])
        matrix[col_num] = column[::-1] #Reverse is the order used in the original code
</pre>